homeassistant/components/ebusd/sensor.py

```diff
@@ -0,0 +1,107 @@
+"""
+Support for Ebusd daemon for communication with eBUS heating systems.
+
+For more details about ebusd deamon, please refer to the documentation at
+https://github.com/john30/ebusd
+"""
+
+import logging
+import datetime
+
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['ebusd']
+
+DATA_EBUSD = 'EBUSD'
+TIME_FRAME1_BEGIN = 'time_frame1_begin'
+TIME_FRAME1_END = 'time_frame1_end'
+TIME_FRAME2_BEGIN = 'time_frame2_begin'
+TIME_FRAME2_END = 'time_frame2_end'
+TIME_FRAME3_BEGIN = 'time_frame3_begin'
+TIME_FRAME3_END = 'time_frame3_end'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Ebus sensor."""
+    ebusd_api = hass.data[DATA_EBUSD]
+    monitored_conditions = discovery_info['monitored_conditions']
+    name = discovery_info['client_name']
+
+    dev = []
+    for condition in monitored_conditions:
+        dev.append(Ebusd(
+            ebusd_api, discovery_info['sensor_types'][condition], name))
+
+    add_entities(dev, True)
+
+
+class Ebusd(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, data, sensor, name):
+        """Initialize the sensor."""
+        self._state = None
+        self._client_name = name
+        self._name = sensor[0]
```

Comment:
I think it would be cleaner/easier to read to do this all at once:

`self._name, self._unit_of_measurement, self._icon, self._type = sensor`